src/slice.rs

```diff
@@ -0,0 +1,123 @@
+//! Non-empty Slices.
+use crate::iter::{IntoNonEmptyIterator, NonEmptyIterator};
+use std::iter::{Chain, Once, Skip};
+
+/// Non-empty slice
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct NESlice<'a, T> {
+    /// The element of the non-empty Vector. Always exists.
+    pub head: &'a T,
+
+    /// The remaining elements of the non-empty Vector, perhaps empty.
+    pub tail: &'a [T],
+}
```

Comment:
I suppose having a macro for this like `nev!` doesn't make much sense eh. If the user wants a macro, they can just use `nev!` for an owned version.